ext/liquid_c/vm.c

```diff
@@ -273,9 +273,10 @@ static VALUE vm_render_until_error(VALUE uncast_args)
                 break;
             case OP_RENDER_VARIABLE_RESCUE:
                 // Save state used by vm_render_rescue to rescue from a variable rendering exception
-                args->node_line_number = (unsigned int)*const_ptr++;
+                memcpy(&args->node_line_number, ip, sizeof(args->node_line_number));
```

Comment:
Nice! Had to check disassembly. I would have assumed `args->node_line_number[0] = ...` would have been better, but it's not.

But it is slower than the previous code.

Reply:
We could try just storing a pointer to the bytes, as I suggested in https://github.com/Shopify/liquid-c/pull/79#discussion_r502504712 .  That seems like it might require less instructions than copying 3 individual bytes.

Reply:
Checked how other VMs do it. I think we should have two instructions: `OP_RENDER_VARIABLE_RESCUE` & `OP_RENDER_VARIABLE_RESCUE_W` (wide version).

The wide version does the decode over the 3 bytes, directly in the `case`, the non-wide one only use one byte (and no bitshifting).

Similar to the `LOADK` & `LOADKX` (wide version) instructions in Lua.  And inspired by Java too, that has many wide (`*_w`) version of instructions: https://en.wikipedia.org/wiki/Java_bytecode_instruction_listings. Also, Java has a whole other category of wide instructions, see the `wide` at the bottom, very interesting.

Reply:
I still think doing any decoding is a waste, since this is just a line number that is only needed on the error path.  I believe MRI ruby and C++ take the approach of keeping the exception handling out of the normal execution path and then just looking up data to handle the rescue path based on the instruction position when an exception is being handled.

The use of the rescue instruction was for simplicity before we need more general rescue handling.

Reply:
I did a quick implementation of `OP_RENDER_VARIABLE_RESCUE` and `OP_RENDER_VARIABLE_RESCUE_W` instruction to benchmark. It didn't seem to perform any faster on the benchmark.

Reply:
I pushed it onto the `pz-render-var-rescue-wide` branch if anyone wants to play around with it.

Reply:
Ah but I see you removed the `memcpy`. It does indeed remove the need for the wide instruction. That's a brilliant idea 👏